drivers/led/led_pwm.c

```diff
@@ -111,12 +103,11 @@ static int led_pwm_init(const struct device *dev)
 
 	for (i = 0; i < config->num_leds; i++) {
 		const struct led_pwm *led = &config->led[i];
+		int ret = device_usable_check(led->dev);
```

Comment:
Here you are using `device_usable_check()`, but in #32560 use are using `device_is_ready()`?

Reply:
I used `device_usable_check` here as this a driver and in #32560 its just a tests/samples so `device_is_ready` felt sufficient there.

Reply:
Are we moving away from device_is_ready in drivers? Why?

Reply:
The concept was that `device_usable_get()` provides standard return values that can be used to determine *why* the device isn't ready, and so allow the device to be made ready and the operation re-tried.

Reply:
I'll let @pabigot chime in here, but I would assume that we'd want to as that would propagate more error information back to the caller.

Reply:
Which is good, but only if the caller intends to try to remedy the device not being ready?

Reply:
If the information isn't available, remedy isn't an option.  But yes, this gets to the gap in functional definition for power management.  If a device call can fail because the device is powered down, this is useful information, as opposed to failing because the device failed a self-test during initialization.

Reply:
@pabigot Certainly. I was talking in the context of the changes done here, where the error is simply reported and nothing else.

I am fine with either solution for checking the device state before use here. I just wondered what made @galak choose two different approaches.

Reply:
OK, so I'm hearing that drivers may use either function depending on what they intend to do with the information. If they're treating a nonready dependency as a fatal error then they may use device_is_ready. Right?

Reply:
> I am fine with either solution for checking the device state before use here. I just wondered what made @galak choose two different approaches.

I made the choice based on if the code was being used in a sample/tests v driver.  Figuring that the sample/test is just reporting a pass/fail state so no need to use `device_usable_check`.  I figured in the driver returning the results of `device_usable_check` to the caller provides more info than just `-ENODEV` if the device isn't available for some other reason.

Reply:
Ok, that's fair. Either that or logging the error code and returning -ENODEV both seem like better approaches. I'll adjust accordingly.

Reply:
That's a useful point.  I was speaking of the use in API functions where the failure value gets sent back to the caller.  There it's not the device that gets to decide whether it's useful, it's the caller.

In an init function, where the failure value propagates to a fatal and permanent device-is-unusable `device_usable_check()` doesn't make as much sense, except to the degree it'd be useful in diagnostics.

Reply:
Sounds good to me, changed the code to use `device_is_ready()`